app/models/asset.rb

```diff
@@ -148,7 +148,7 @@ class VolumeError < StandardError
   # We accept not only an individual barcode but also an array of them.
   scope :with_barcode, ->(*barcodes) {
     db_barcodes = Barcode.extract_barcodes(barcodes)
-    includes(:barcodes).where(barcodes: { barcode: db_barcodes }).distinct
+    joins(:barcodes).where(barcodes: { barcode: db_barcodes }).distinct
   }
```

Comment:
This is the main fix, with a similar change elsewhere:
1) By default rails converts an include to a 'preload' which essentially results in separate queries.
2) However if there is a condition that covers the include, rails converts it to a join.
3) Unfortunately rails also converts ALL other includes to joins. Which is a problem if you have lots of aliquots and requests.
4) Perhaps unsurprisingly rails is actually in the process of adjusting this behaviour, forcing you to be a bit more explicit. I *think* the previous behaviour we had is actually deprecated in the latest version, although some of the discussion I could find was slightly contradictory.